analyzers/tests/ITs.JsonParser.Test/JsonParsingTest.cs

```diff
@@ -1,12 +1,187 @@
+
+        VerifyResultFile(outFile1, """
+            {
+              "Issues": [
+                {
+                  "Id": "S100",
+                  "Message": "Message_1.",
+                  "Location": {
+                    "Uri": "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/foo/bar/File1.cs#L1",
+                    "Region": {
+                      "StartLine": 1,
+                      "StartColumn": 42,
+                      "EndLine": 1,
+                      "EndColumn": 69
+                    }
+                  }
+                },
+                {
+                  "Id": "S100",
+                  "Message": "Message_2.",
+                  "Location": {
+                    "Uri": "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/foo/bar/File1.cs#L42-L43",
+                    "Region": {
+                      "StartLine": 42,
+                      "StartColumn": 42,
+                      "EndLine": 43,
+                      "EndColumn": 69
+                    }
+                  }
+                }
+              ]
+            }
+            """);
+
+        VerifyResultFile(outFile2, """
+            {
+              "Issues": [
+                {
+                  "Id": "S142",
+                  "Message": "Message_3.",
+                  "Location": {
+                    "Uri": null,
+                    "Region": null
+                  }
+                }
+              ]
+            }
+            """);
+
+        VerifyResultFile(outFile3, """
+            {
+              "Issues": [
+                {
+                  "Id": "S100",
+                  "Message": "Message_1.",
+                  "Location": {
+                    "Uri": "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/foo/bar/File1.cs#L1",
+                    "Region": {
+                      "StartLine": 1,
+                      "StartColumn": 42,
+                      "EndLine": 1,
+                      "EndColumn": 69
+                    }
+                  }
+                }
+              ]
+            }
+            """);
+
+        VerifyResultFile(outFile4, """
+            {
+              "Issues": [
+                {
+                  "Id": "S100",
+                  "Message": "Message_1.",
+                  "Location": {
+                    "Uri": "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/foo/bar/File1.cs#L1",
+                    "Region": {
+                      "StartLine": 1,
+                      "StartColumn": 42,
+                      "EndLine": 1,
+                      "EndColumn": 69
+                    }
+                  }
+                }
+              ]
+            }
+            """);
+    }
+
+    private string TestDirectory()
+    {
+        var root = Path.Combine(TestContext.TestRunDirectory, TestContext.TestName);
+        Directory.CreateDirectory(root);
+        return root;
```

Comment:
Use the TestPath helper here. It will make sure that the directories will not end up all over the place. We had issues that those path are not super-consistent across different TFMs and versions